provider/bikeshed/src/units/essentials.rs

```diff
@@ -42,13 +45,64 @@ impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
             .ok_or_else(|| DataError::custom("Failed to get times"))?
             .clone();
 
-        // TODO: Fill prefixes (power, si prefixes, ... etc.) in the next PR.
-        let prefixes = ZeroMap::new();
+        // TODO: Fill prefixes (si prefixes, ... etc.) in the next PR.
+        let mut prefixes = BTreeMap::<PatternKey, String>::new();
+        fn fill_power(
+            prefixes: &mut BTreeMap<PatternKey, String>,
+            powers: Option<&Patterns>,
+            power_value: PowerValue,
+        ) {
```

Comment:
nit: use a for loop

Reply:
what do you mean? I am already using a for loop to insert counts.

Reply:
instead of defining a method and callling it twice

```rust
        for (powers, power_value) in [
            (length_data.get("power2"), PowerValue::Two),
            (length_data.get("power3"), PowerValue::Three),
        ] {
           // the content of this function
        }
```

Reply:
done, I split it into different function to remove extra nested loop

Reply:
Removing nested loops is not the goal here. Having nested functions that directly get called really hurts readability, please just inline them.

Reply:
done